tools/common/common-build.targets

```diff
@@ -9,24 +9,24 @@
     <PackageContentBasePath Condition="'$(ShouldAssignPackageContentBasePath)' == 'true'">build\lib\$(TargetOsAndVersion)\$(PlatformTarget)</PackageContentBasePath>
     <PackageContentBasePath Condition="'$(ShouldAssignPackageContentBasePath)' == 'true' and '$(HasConfigurationSpecificPackageContents)' == 'true'">$(PackageContentBasePath)\$(Configuration)</PackageContentBasePath>
   </PropertyGroup>
+  <Target Name="AddBuiltOutput" BeforeTargets="GetPackageContents"  DependsOnTargets="BuiltProjectOutputGroup;$(GetPackageContentsDependsOn)" Condition="'$(ConfigurationType)' == 'DynamicLibrary' or '$(ConfigurationType)' == 'Application'">
```

Comment:
>BuiltProjectOutputGroup [](start = 86, length = 23)

This change scares me a little. Its probably fine but there were issues in the past where we would link multiple times per flavor (config+platform) causing issues (symbols didn't match binaries) because of the dependencies of packaging. Can you validate that in an AnyCPU build we only link once per flavor?

Reply:
In general, the idea was that just building a .vcxproj itself would only build the outputs. Building a .nuproj would only package it and its dependencies up. This means that the AddBuiltOutput target should only be run if the build already happened (which it will have because of project dependencies such that the .nuproj is only built after its dependencies, the .vcxprojs, are built).

All of the above is totally fine if the BuiltProjectOutputGroup is only used to figure out what the project outputs are and where they will end up, not actually to produce them. 

---
In reply to: [141685886](https://github.com/Microsoft/WinObjC/pull/2799#discussion_r141685886) [](ancestors = 141685886)

Reply:
You can double check that this is the case by looking at Microsoft.CppBuild.targets or wherever its defined and checking the dependencies of this target.

---
In reply to: [141687222](https://github.com/Microsoft/WinObjC/pull/2799#discussion_r141687222) [](ancestors = 141687222,141685886)

Reply:
The problem was that various dll's weren't being included in the final output package. This seems to fix the issue but I did want to talk to you about it.